### PR TITLE
Use XDG_RUNTIME_DIR for IPC socket (Bug #1353)

### PR DESCRIPTION
**Pull request by saivert**
Fixes bug #1353

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -880,6 +881,17 @@ main (int argc, char *argv[]) {
         fprintf (stderr, "fatal: too long cache path %s\n", dbcachedir);
         return -1;
     }
+
+    const char *xdg_runtime = getenv("XDG_RUNTIME_DIR");
+    if (!xdg_runtime) {
+        fprintf (stderr, "Warning: XDG_RUNTIME_DIR not set falling back to /tmp/deadbeef\n");
```
Comment (Alexey-Yakovenko) [inline main.c]: wouldn't this be printing this warning on pretty much every computer?

File: main.c [hunk 2]
```diff
@@ -880,6 +881,17 @@ main (int argc, char *argv[]) {
...
+    }
+    const char *runtime_root = xdg_runtime ? xdg_runtime : "/tmp";
+    if (snprintf(dbruntimedir, sizeof (dbruntimedir), xdg_runtime ? "%s/deadbeef/" : "%s/deadbeef/", runtime_root) >= sizeof (dbruntimedir)) {
```
Comment (Alexey-Yakovenko) [inline main.c]: this will break when 2 users try to use deadbeef at the same time

File: main.c [hunk 3]
```diff
@@ -880,6 +881,17 @@ main (int argc, char *argv[]) {
         fprintf (stderr, "fatal: too long cache path %s\n", dbcachedir);
         return -1;
     }
+
+    const char *xdg_runtime = getenv("XDG_RUNTIME_DIR");
+    if (!xdg_runtime) {
+        fprintf (stderr, "Warning: XDG_RUNTIME_DIR not set falling back to /tmp/deadbeef\n");
+    }
+    const char *runtime_root = xdg_runtime ? xdg_runtime : "/tmp";
```
Comment (Alexey-Yakovenko) [inline main.c]: this (/tmp/deadbeef) will break when 2 users try to use deadbeef at the same time

File: main.c [hunk 4]
```diff
@@ -92,6 +92,7 @@ char dbdocdir[PATH_MAX]; // see deadbeef->get_doc_dir
 char dbplugindir[PATH_MAX]; // see deadbeef->get_plugin_dir
 char dbpixmapdir[PATH_MAX]; // see deadbeef->get_pixmap_dir
 char dbcachedir[PATH_MAX];
+char dbruntimedir[PATH_MAX]; // /run/user/<uid>/deadbeef
```
Comment (Alexey-Yakovenko) [inline main.c]: this should be a local variable -- it's not needed anywhere else.

Comment (Alexey-Yakovenko) [inline main.c]: (nvm the last comment, deleted)

**Comment by Alexey-Yakovenko** [comment]
what should happen instead:

Check if XDG_RUNTIME_DIR is set, and use it. Otherwise keep the old behavior.

Printing the warning for 99.9% users is annoying, so please don't do that (yeah, I know it's recommended in the FDO paper -- they're wrong).

**Comment by saivert** [comment]
Better?

File: main.c [hunk 5]
```diff
@@ -977,7 +991,7 @@ main (int argc, char *argv[]) {
     len = offsetof(struct sockaddr_un, sun_path) + sizeof (server_id)-1;
 #else
     char *socketdirenv = getenv ("DDB_SOCKET_DIR");
-    snprintf (remote.sun_path, sizeof (remote.sun_path), "%s/socket", socketdirenv ? socketdirenv : dbconfdir);
+    snprintf (remote.sun_path, sizeof (remote.sun_path), "%s/socket", socketdirenv ? socketdirenv : dbruntimedir);
```
Comment (Alexey-Yakovenko) [inline main.c]: looks like the dbruntimedir variable will get used here before being initialized.

Comment (Alexey-Yakovenko) [inline main.c]: I'm wrong, nvm.

**Comment by Alexey-Yakovenko** [comment]
I think we should just use this:
```
--- a/main.c
+++ b/main.c
@@ -402,7 +402,7 @@ server_start (void) {
     memcpy (srv_local.sun_path, server_id, sizeof (server_id));
     int len = offsetof(struct sockaddr_un, sun_path) + sizeof (server_id)-1;
 #else
-    char *socketdirenv = getenv ("DDB_SOCKET_DIR");
+    char *socketdirenv = getenv ("XDG_RUNTIME_DIR");
     snprintf (srv_local.sun_path, sizeof (srv_local.sun_path), "%s/socket", socketdirenv ? socketdirenv : dbconfdir);
     if (unlink(srv_local.sun_path) < 0) {
         perror ("INFO: unlink socket");
@@ -976,7 +976,7 @@ main (int argc, char *argv[]) {
     memcpy (remote.sun_path, server_id, sizeof (server_id));
     len = offsetof(struct sockaddr_un, sun_path) + sizeof (server_id)-1;
 #else
-    char *socketdirenv = getenv ("DDB_SOCKET_DIR");
+    char *socketdirenv = getenv ("XDG_RUNTIME_DIR");
     snprintf (remote.sun_path, sizeof (remote.sun_path), "%s/socket", socketdirenv ? socketdirenv : dbconfdir);
     len = offsetof(struct sockaddr_un, sun_path) + strlen (remote.sun_path);
 #endif
```

**Comment by saivert** [comment]
Yes except it has to create a directory for the socket file otherwise the socket file ends up in XDG_RUNTIME_DIR dir.
Unless you rename it to deadbeefsocket or something. I think it is a lot cleaner with a separate directory for deadbeef though.

**Comment by Alexey-Yakovenko** [comment]
haha yeah that's what I meant by "or similar" after the edit. anyway, let's keep it the way you made it for now.

merged.

**Pull request closed without merging**